lib/fluent/plugin/exception_detector.rb

```diff
@@ -360,8 +362,14 @@ def add(time_sec, record, message)
         @buffer_start_time = Time.now
       end
       unless message.nil?
-        @messages << message
-        @buffer_size += message.length
+        message_with_line_break =
+          if @add_line_break && !@messages.empty?
+            "\n" + message
```

Comment:
We should also check if the message already ends with a `\n`, right? Basically we should only append `\n` if a message is not ending with `\n`.